healthchecks/healthchecks.go

```diff
@@ -68,3 +68,42 @@ func runCheckUntilSuccess(host Host, healthCheck HealthCheck, wg *sync.WaitGroup
 	}
 	wg.Done()
 }
+
+func PerformPreChecks(sshContext *ssh.SSHContext, host Host, timeout int) (err error) {
```

Comment:
Let's try to merge this with `Perform`, and along with the type change we should get http checks as pre-checks for free